src/hooks/useUser.ts

```diff
@@ -1,29 +1,44 @@
-import { User, UseUserReturnType } from '@/interfaces';
-import { useSession } from 'next-auth/react';
 import { useState, useEffect } from 'react';
+import { useSession } from 'next-auth/react';
+import { User, UseUserReturnType } from '@/interfaces';
 
 const useUser = (): UseUserReturnType => {
   const { data: session, status } = useSession();
-  const [user, setUser] = useState<User>(session?.user as User);
+  const [user, setUser] = useState<User | null>(null);
   const [isAuth, setIsAuth] = useState(false);
+  const [isOnboarded, setIsOnboarded] = useState<boolean | undefined>(
+    undefined
+  );
   const [loading, setLoading] = useState(true);
 
   useEffect(() => {
     if (status === 'loading') {
       setLoading(true);
-    } else {
-      setLoading(false);
+      return;
     }
 
+    setLoading(false);
+
     if (session?.user) {
-      setUser(session?.user as User);
+      const sessionUser = session.user as User;
+      const onboarded = sessionUser.isOnboarded ?? false;
+      setUser({
```

Comment:
can't you write like - 

{
        id,
        name,
        emai,
        image,
        isOnboarded,
      }

by restructuring